framework/components/blockchain/containers.go

```diff
@@ -0,0 +1,75 @@
+package blockchain
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/go-connections/nat"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/framework"
+)
+
+func baseRequest(in *Input) testcontainers.ContainerRequest {
+	containerName := framework.DefaultTCName("blockchain-node")
+	bindPort := fmt.Sprintf("%s/tcp", in.Port)
+
+	return testcontainers.ContainerRequest{
+		Labels:       framework.DefaultTCLabels(),
+		Name:         containerName,
+		ExposedPorts: []string{bindPort},
+		HostConfigModifier: func(h *container.HostConfig) {
+			h.PortBindings = framework.MapTheSamePort(bindPort)
+			framework.ResourceLimitsFunc(h, in.ContainerResources)
+		},
+		Networks: []string{framework.DefaultNetworkName},
+		NetworkAliases: map[string][]string{
+			framework.DefaultNetworkName: {containerName},
+		},
+		WaitingFor: wait.ForListeningPort(nat.Port(in.Port)).WithStartupTimeout(10 * time.Second).WithPollInterval(200 * time.Millisecond),
+	}
+}
+
+func createGenericEvmContainer(in *Input, req testcontainers.ContainerRequest) (*Output, error) {
```

Comment:
is it generic enough to use with all the other EVM types we support? if so, maybe we should reuse it there?

Reply:
Let me check...

Reply:
I refactored the EVM ones. Should I work with the others too?

Reply:
no need, thanks!